Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -843,8 +843,39 @@ open class NavigationMapView: UIView {
         mapView.style.removeSources(sourceSet)
     }
     
+    /**
+     Attempts to localize road labels into the local language and other labels into the system’s preferred language.
+     
+     This method automatically modifies the `SymbolLayer.Layout.textField` property of any symbol style layer whose source is the <a href="https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/#overview">Mapbox Streets source</a>. The user can set the system’s preferred language in Settings, General Settings, Language & Region.
+     
+     This method localizes road labels into the local language, regardless of the system’s preferred language, in an effort to match road signage. The turn banner always displays road names and exit destinations in the local language, so if this `NavigationMapView` stands alone outside a you should call the `BaseMapView.on(_:handler:)` on `mapView`, passing in `MapEvents.EventKind.styleLoaded`, and call this method inside the closure. The map view embedded in `NavigationViewController` is localized automatically, so you do not need to call this method on the value of `NavigationViewController.navigationMapView`.
+     */
     public func localizeLabels() {
-        // TODO: Implement ability to localize road labels.
+        guard !AccountManager.hasChinaBaseURL,
+              let layers = try? mapView.style.styleManager.getStyleLayers() else {
+            return
+        }
+        
+        let streetsSourceIdentifiers: [String] = mapView.streetsSources().map { $0.id }
+        
+        for layerInfo in layers where layerInfo.type == "symbol" {
+            guard var layer = try? mapView.style.getLayer(with: layerInfo.id, type: SymbolLayer.self).get(),
```

Comment:
This call fails on styles that set `text-field` to a formatted expression, such as Streets and Navigation Day: mapbox/mapbox-maps-ios#268. Unfortunately, the label localization feature has no effect in these styles.

Reply:
Per https://github.com/mapbox/mapbox-maps-ios/issues/268#issuecomment-820317143, the bug is fixed in v2.0.0-beta.17.